### DISCUSSION TITLE
Custom Input Rule always removing the last char

### DISCUSSION THREAD

**Discussion by Andre2553** [ueberdosis/tiptap#3879] [General Questions]
Hi guys, I am working on a new mark, and I am having a lot of problems achieving what I want.

I want to have an input rule that is triggered where I have an open curly bracket and closed curly bracket e.g. {test}, but I want to keep the curly brackets when the mark is created. So, it will stay {test}

But the default rule deletes the first char and the last one, leaving just the test. 
I was able to keep the first one, but I don't know how to keep the last {test

this is is the code
`const inputRegex = /{[^{}]*}/;

const CustomSpan = Mark.create({
    name: 'customSpan',
    selectable: false,
    group: 'inline',
    inclusive: false,
    inline: true,
    atom: true,
    addOptions() {
        return {
            HTMLAttributes: {},
        }
    },
    parseHTML() {
        return [{ tag: 'span' }]
    },
    renderHTML({ HTMLAttributes }) {
        return ['span', mergeAttributes(this.options.HTMLAttributes, HTMLAttributes), 0]
    },
    
    addInputRules() {
        return [
            markInputRule({
                find: inputRegex,
                type: this.type,
            }),
        ];
    },

})`

**Comment by GabrielJuradoCarsync** [comment]
Hi, did you found a way to fix this?

**Reply by Abacaxi-Nelson** [reply]
Same problem here, any solution ?
